### PR TITLE
ccminer mac

### PR DESCRIPTION
**Pull request by kangcliff**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -91,6 +91,12 @@ ccminer_LDFLAGS  = $(PTHREAD_FLAGS) @CUDA_LDFLAGS@
 ccminer_LDADD    = @LIBCURL@ @JANSSON_LIBS@ @PTHREAD_LIBS@ @WS2_LIBS@ @CUDA_LIBS@ @OPENMP_CFLAGS@ @LIBS@ $(nvml_libs)
 ccminer_CPPFLAGS = @LIBCURL_CPPFLAGS@ @OPENMP_CFLAGS@ $(CPPFLAGS) $(PTHREAD_FLAGS) -fno-strict-aliasing $(JANSSON_INCLUDES) $(DEF_INCLUDES) $(nvml_defs)
 
+if HAVE_OSX
+ccminer_LDFLAGS += -L/usr/local/opt/llvm/lib
+ccminer_LDADD += -lcurl -lcrypto -lomp
+ccminer_CPPFLAGS += -I/usr/local/opt/openssl/include -I/usr/local/llvm/lib/clang/4.0.0/include
```
Comment (tpruvot) [inline Makefile.am:97]: there is no pkgconfig on mac ?

File: scrypt.cpp [hunk 2]
```diff
@@ -43,7 +43,11 @@ using namespace Concurrency;
 #include <string.h>
 
 #include <emmintrin.h>
+#if !((defined(__FreeBSD__) && __FreeBSD__ >= 10) || defined(__APPLE__))
```
Comment (tpruvot) [inline scrypt.cpp:46]: im not fan of these reversed if else

File: scrypt/salsa_kernel.h [hunk 3]
```diff
@@ -3,7 +3,11 @@
 
 #include <stdio.h>
 #include <stdbool.h>
+#if !((defined(__FreeBSD__) && __FreeBSD__ >= 10) || defined(__APPLE__))
```
Comment (tpruvot) [inline scrypt/salsa_kernel.h:6]: same

File: configure.ac [hunk 4]
```diff
@@ -88,8 +96,15 @@ else
...
 fi
 
-LIBCURL_CHECK_CONFIG(, 7.15.2, ,
-  [AC_MSG_ERROR([Missing required libcurl >= 7.15.2])])
+#LIBCURL_CHECK_CONFIG(, 7.15.2, ,
+#  [AC_MSG_ERROR([Missing required libcurl >= 7.15.2])])
+LIBCURL=-lcurl
+if test x$have_osx = xtrue
+then
+  LIBCURL_CPPFLAGS=-I/usr/local/opt/curl/include
```
Comment (tpruvot) [inline configure.ac:104]: its supposed to use libcurl.pc in pkgconfig folder for the path (which doesnt look normal here)

**Review by tpruvot — CHANGES_REQUESTED** [review]

File: Makefile.am (hunk 1 above)
Comment (kangcliff) [reply Makefile.am:97]: There is, but the llvm package I installed doesn't come with .pc files.

File: scrypt.cpp (hunk 2 above)
Comment (kangcliff) [reply scrypt.cpp:46]: changed to #ifdef __APPLE__

**Comment by kangcliff** [comment]
Benchmark results on GTX 1070 with stock settings. 

Nvidia Web Driver 378.05.05 with CUDA 8.0.61:
[ccminer_bench_mac.txt](https://github.com/tpruvot/ccminer/files/981659/ccminer_bench.txt)

Geforce Game Ready Driver 382.05 with CUDA 7.5:
[ccminer_bench_win.txt](https://github.com/tpruvot/ccminer/files/981654/ccminer_bench.txt)

File: INSTALL [hunk 5]
```diff
@@ -40,3 +40,19 @@ and comment/delete the line 82 : #error -- unsupported GNU version! gcc 4.9
...
 
+** How to compile on macOS **
+# Step 1: download and install CUDA Toolkit
+# https://developer.nvidia.com/cuda-toolkit-archive
+
+# Step 2: install Homebrew
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+
+# Step 3: dependencies
+brew install autoconf automake curl openssl llvm
```
Comment (tpruvot) [inline INSTALL:51]: could you try if "brew install pkg-config" works ?

Comment (kangcliff) [reply INSTALL:51]: I installed pkg-config with brew already, but I cannot get include paths using these:

$(shell pkg-config --cflags libcurl)
$(shell pkg-config --cflags libcrypto)

**Comment by tpruvot** [comment]
I made a commit which "merge" a part of these changes : https://github.com/tpruvot/ccminer/commit/fad5b0470a559b170f30433c4b730c7c53c6c216

**Comment by nikbass** [comment]
hi. news for this project? 
I'm trying to compile it from a couple of days on my mac pro but without success. Unfortunately, I have no experience
thx advance.

**Comment by tmikaeld** [comment]
I'm trying to compile this on High Sierra, but setting the compiler flags doesn't help configure to find the curl sources - it does find openssl though.

My FLAGS in build.sh

```bash
CPPFLAGS="-I/usr/local/opt/openssl/include -I/usr/local/opt/curl/include" \
LDFLAGS="-L/usr/local/opt/openssl/lib -L/usr/local/opt/curl/lib" \
./configure.sh
```

The configure error:

```bash
checking for pthread_create in -lpthread... yes
checking for gzopen in -lz... yes
checking for SSL_free in -lssl... yes
checking for EVP_DigestFinal_ex in -lcrypto... yes
./configure: line 6165: syntax error near unexpected token `,'
./configure: line 6165: `LIBCURL_CHECK_CONFIG(, 7.15.2, ,

**Comment by rooi** [comment]
@tmikaeld have you tried running the following?

`brew link curl --force`

I've written down my actions to build and use tpruvot's ccminer on: https://github.com/rooi/ccminer_macos/README.md

**Pull request closed without merging**